src/components/Player/ProfileHeaderInfo.svelte

```diff
@@ -61,10 +62,15 @@
 
   let nameInput;
   let redactingName = false;
-  function onRedactButtonClick() {
-      if (redactingName && nameInput.value) {
-          account.changeName(nameInput.value);
+  async function onRedactButtonClick() {
+      if (!redactingName) nameInput = name;
+
+      if (redactingName && nameInput !== name) {
+          await account.changeName(nameInput);
+
+          dispatch('player-data-updated', {name: nameInput})
```

Comment:
A name change can potentially fail here, need to check the `account.error`, or maybe return the error from `account.changeName` function.

Reply:
Ok, I found it. I think error handling should be standardized too. Now it just returns a string. IMHO it should be JSON with a fixed format.

Reply:
I see it's done 😁
![image](https://user-images.githubusercontent.com/1758707/158170633-d7a1ef92-a5b5-4ddb-b58b-63b54cb82b25.png)

Reply:
Hmm, not yet xD
![image](https://user-images.githubusercontent.com/1758707/158170803-42751841-66cb-457e-8b10-4dcb154dbd2a.png)

Reply:
Ok, it's done.

![image](https://user-images.githubusercontent.com/1758707/158171336-a5a2ca5f-3041-47b8-a0d4-07fe305b2b08.png)